zerver/lib/markdown/fenced_code.py

```diff
@@ -215,6 +246,39 @@ def check_for_new_fence(
         output.append(line)
 
 
+class KrokiHandler(BaseHandler):
+    def __init__(
+        self,
+        processor: Any,
+        output: MutableSequence[str],
+        fence: str,
+        lang: str,
+    ) -> None:
+        self.processor = processor
+        self.output = output
+        self.fence = fence
+        self.lang = lang
+        self.lines: List[str] = []
+
+    def handle_line(self, line: str) -> None:
+        self.lines.append(line)
+
+    def done(self) -> None:
+        text = "\n".join(self.lines)
+        processed_lines = text.rstrip()[:-3].rstrip()
+
+        encoded_diagram = base64.urlsafe_b64encode(
+            zlib.compress(bytes(processed_lines, "utf-8"), 9)
+        ).decode("utf-8")
+
+        kroki_url = settings.KROKI_RENDERING_URL.format(
+            self.lang.replace("kroki ", ""), encoded_diagram
+        )
+        final_text = self.processor.placeholder(requests.get(kroki_url).text)
```

Comment:
This will need to be sandboxed. I think we want to generate an `<img>` tag pointing to the SVG URL.